feat: Annotate Guild Territory Upgrades

cmkohnen opened the pull request:
See the following example:
![Bildschirmfoto vom 2024-05-20 20-04-37](https://github.com/Wynntils/Artemis/assets/59118042/3855a920-e436-4614-9f04-d36794e8a22b)

kristofbolyai commented on common/src/main/java/com/wynntils/models/items/ItemModel.java:
```diff
@@ -34,18 +34,8 @@
-import com.wynntils.models.items.annotators.gui.SkillPointAnnotator;
-import com.wynntils.models.items.annotators.gui.SoulPointAnnotator;
-import com.wynntils.models.items.annotators.gui.TerritoryAnnotator;
+import com.wynntils.models.items.annotators.gui.*;
```
Please configure your Artemis IDEA project settings to not use wildcard imports (and static imports).

cmkohnen committed:
feat: Annotate Bonus/Tower Items

Annotate items in Guild Territory Management to easier differentiate
between upgrades at lvl. 1 vs. upgrades at lvl. 0

Signed-off-by: Christoph Matthias Kohnen <christoph.kohnen@disroot.org>

kristofbolyai commented on common/src/main/java/com/wynntils/models/items/annotators/gui/GuildTerritoryUpgradeAnnotator.java:
```diff
@@ -0,0 +1,30 @@
+import java.util.regex.Pattern;
+import net.minecraft.world.item.ItemStack;
+
+public class GuildTerritoryUpgradeAnnotator implements GuiItemAnnotator {
```
Can you rename this to `TerritoryUpgradeItem` so it matches `TerritoryItem`'s naming?

ShadowCat117 replied:
And make sure to reorder it in `ItemModel` so it's still alphabetical :)

kristofbolyai commented on common/src/main/java/com/wynntils/models/items/annotators/gui/GuildTerritoryUpgradeAnnotator.java:
```diff
@@ -0,0 +1,30 @@
+import net.minecraft.world.item.ItemStack;
+
+public class GuildTerritoryUpgradeAnnotator implements GuiItemAnnotator {
+    private static final Pattern GUILD_TERRITORY_UPGRADE_PATTERN =
```
And since you already had such nice test cases, can you add a new test in TestRegex.java? (Optional)

cmkohnen committed:
refactor: Adress review

Signed-off-by: Christoph Matthias Kohnen <christoph.kohnen@disroot.org>

kristofbolyai reviewed (approved):
Thank you for adding the tests!
